libs/sdk-ui-dashboard/src/examples.tsx

```diff
@@ -0,0 +1,108 @@
+// (C) 2019 GoodData Corporation
+import { MenuButtonItem } from "./topBar";
+import React from "react";
+import { IDashboardAttributeFilterProps } from "./filterBar";
+import { DashboardWidget } from "./dashboard/DashboardWidget";
+import { idRef } from "@gooddata/sdk-model";
+import { InsightView } from "@gooddata/sdk-ui-ext";
+import { Dashboard } from "./dashboard";
+
+/**
+ * Shows how to use built-in customization to influence placement of the menu
+ */
+export function demonstrateCustomMenuPlacement() {
+    return (
+        <Dashboard
+            topBarConfig={{
+                defaultComponentProps: {
+                    menuButtonConfig: {
+                        placement: "left",
+                    },
+                },
+            }}
+        />
+    );
+}
+
+/**
+ * Shows how to use built-in customization to influence placement and add custom menu items
+ */
+export function demonstrateCustomMenuPlacementAndItem() {
+    const customItem: [number, MenuButtonItem] = [
+        -1,
+        {
+            itemId: "myItem",
+            itemName: "Open My App",
+            callback: () => {
+                window.location.assign("/bang");
+            },
+        },
+    ];
+    return (
+        <Dashboard
+            topBarConfig={{
+                defaultComponentProps: {
```

Comment:
Render props are imho better pattern for these customizations.

Reply:
yep we definitely have to revisit this. this was some initial idea that i sketched. as we add the real component impls, we will see.

the idea behind this setup was to allow simple, easy, almost declarative-like customization of the default implementations